include/dynd/types/type_id.hpp

```diff
@@ -175,6 +175,35 @@ enum type_id_t {
   static_id_max = dim_fragment_id
 };
 
+enum type_property_kind_t {
+  Int64_kind,
+  Uint64_kind,
+  Size_kind,
+  Intptr_kind,
+  Uintptr_kind,
+  String_kind,
+  Type_kind,
+  UintptrVector_kind,
+  StringVector_kind,
+  TypeVector_kind
+};
+
+typedef struct {
+  type_property_kind_t kind;
```

Comment:
Here we could in fact use type_id_t, but: Some types are missing, and we would need a reserved flag (something like the high bit of a size_t, if the type_id_t enum is changed to size_t) to distinguish between vectors and plain types.